py/desispec/ccdcalib.py

```diff
@@ -423,28 +460,38 @@ def compute_nightly_bias(night, cameras, outdir=None, nzeros=25, minzeros=20,
         rank, size = 0, 1
 
     #- Find all zeros for the night
-    expids = None
+    expdict = None
     if rank == 0:
-        expids = _find_zeros(night)
+        expdict = _find_zeros(night,cameras=cameras,minzeros=nzeros)
 
     if comm is not None:
-        expids = comm.bcast(expids, root=0)
+        expdict = comm.bcast(expdict, root=0)
+
+    rawfiles_dict={}
+    skipped_cams=[]
```

Comment:
You could tuck all of lines 471-493 under line 465 (the `_find_zeros()` line) within the `if rank==0`. Then broadcast the shortened `expdict` to all ranks assuming you didn't exit with no valid cameras.

Otherwise you're broadcasting a larger dictionary and having all ranks calculate all cams.

Reply:
done, also saved the skipped cameras by just removing from the expdict and checking that later